components/automate-cli/cmd/chef-automate/deployedConfig.go

```diff
@@ -0,0 +1,303 @@
+package main
+
+import (
+	"errors"
+
+	"github.com/chef/automate/lib/config"
+)
+
+func PopulateHaCommonConfig(configPuller PullConfigs) (haDeployConfig *config.HaDeployConfig, err error) {
```

Comment:
Can you provide interface for these public functions to mock them?

Reply:
We can directly call PopulateHaCommonConfig() by mocking PullConfigs() which is an interface already for tests as i've done in . I don't see any need to make PopulateHaCommonConfig() an interface.